File: data/CodeLlama/readable_format_HPEC/cpp/file067.cpp
#include <iostream>
#include <omp.h>

int main() {
    omp_lock_t lck;
    int var = 0;

    omp_init_lock(&lck);

    #pragma omp target map(tofrom:var) device(0)
    #pragma omp teams distribute reduction(+:var)
    for (int i = 1; i <= 100; ++i) {
        omp_set_lock(&lck);
        var = var + 1;
        omp_unset_lock(&lck);
    }

    omp_destroy_lock(&lck);

    std::cout << var << std::endl;

    return 0;
}

File: data/deepseek_coder_base_6.7B/readable_format_HPEC/fortran/file067.f90
program DRB154_missinglock3_orig_gpu_no
    use omp_lib
    implicit none

    integer (kind=omp_lock_kind) :: lck
    integer :: var, i
    var = 0

    call omp_init_lock (lck)

    !$omp target map(tofrom:var) device(0)
    !$omp teams distribute reduction(+:var)
    do i = 1, 100
        call omp_set_lock(lck)
        var = var+1
        call omp_unset_lock(lck)
    end do
    !$omp end teams distribute
    !$omp end target

    call omp_destroy_lock(lck)

    print*, var
end program
